### ISSUE TITLE
Training condition (learning speed, learning pattern) is quite different in the latest version(v0.10.0)

### ISSUE THREAD

**Issue by hanseoup** [Unity-Technologies/ml-agents#2713]
Hello.
I found strange points in "ml-agent", after I made version up from `0.9.3` to `0.10.0`. 
I tried to train "my custom battle environment" and I compared its win rate in same step(about 5.0M). But It showed quite different win rate (91% 75%). I didn't do anything except changing ml-agent version! 

My team Leader also felt strange about this issue, so he had checked the sample environment "Hallway". And he found the learning pattern, speed are quite different between version 0.8.0 and version 0.10.0 

Here is the graph with different ml-agent version. 
orange: ml-agent 0.8.0
gray: ml-agent 0.9.3
green: ml-agent 0.10.0
![image](https://user-images.githubusercontent.com/53593292/66536569-63787980-eb58-11e9-832a-d76ee7d7b189.png)

0.8.0 version learned well, but other versions not.
Is there any problem in algorithm of updated version?
Could you explained this? :(

**Comment by hanseoup** [comment]
![image](https://user-images.githubusercontent.com/53593292/66539717-071b5700-eb64-11e9-8dcb-4b785e2efe47.png)

I've retrained hallway example with `0.10.0` about 200K (Blue)
It still seems not to start learning, but it take much longer time than `0.8.0` (about x10)
Could you check the hallway example on your machine with 0.10.0?
And please tell me when the graph is converged. 
Thank you :)

**Comment by ervteng** [comment]
HI @hanseoup, are you using multiple environments by any chance? We changed the way the steps are logged in 0.9 - previously, each step is across all environments (e.g. 1 step with 8 environments would actually be 8 steps), which is now corrected. 

Also, your curve in orange is much steeper than I've seen for Hallway in any version. There's a bit of luck involved - training hallway multiple times can result in different curves. Can you post your hyperparams as well? Thanks!

**Comment by nomoreid** [comment]
we tested multiple times. multiple computer. multiple version. default setting. (3 times on 0.8 vs 0.10)
version 0.8 (default setting) always learning very well on hallway example. (10-20 min on unity editor)
but version 0.10 is not so good(more then 2 hour on unity editor)
we use default hyper-params. it's not luck.  
may be some problem on hallway env or training code.

**Comment by hanseoup** [comment]
Yes. we just used `trainer_config.yaml` file included in "config" folder.
And we just used specific version of `trainer_config.yaml` file. 
So, It could be slightly different by its version.

**Comment by xiaomaogy** [comment]
@ervteng

**Comment by ervteng** [comment]
Hi @hanseoup, @nomoreid, I did some runs of Hallway on my end. They all roughly train in similar time - though, as I mentioned before, there is a bit of randomness in the runs. Green is 0.8, pink is 0.9, and red is 0.10 with the 0.8 Hallway env. 

Note that I do expect the 0.10 training run to be different as the Hallway env (black theme) was changed in 0.10. But the trainer should behave the same way across all the versions. 

![image](https://user-images.githubusercontent.com/5085265/66785481-3e567480-ee92-11e9-94a7-0a9a51d7fc75.png)

Could you by any chance share the hyperparameters used in your custom environment? The one thing that changed a lot between 0.8 and 0.9.3 was the curiosity module.

**Comment by hanseoup** [comment]
Sure. Here it is.

simpleBattleLearning:
    trainer: ppo
    batch_size: 256
    beta: 5.0e-3
    buffer_size: 4096
    epsilon: 0.2
    gamma: 0.99
    hidden_units: 128
    lambd: 0.95
    learning_rate: 3.0e-4
    max_steps: 1.0e7
    memory_size: 64
    normalize: true
    num_epoch: 3
    num_layers: 3
    time_horizon: 64
    sequence_length: 4
    summary_freq: 1000
    use_curiosity: false
    curiosity_strength: 0.01
    curiosity_enc_size: 128
    use_recurrent: false

And we found that hallway learning is stable after I made version up from 0.10.0 to 0.10.1
also, my environment learning comes stable too. 

I saw the quite a lot of hot-fix in 0.10.1 about 5 days ago. 
I think there were problem solving fix in its hot-fix.

**Comment by ervteng** [comment]
Hi @hanseoup, thanks for sending the params and glad 10.1 fixed the issue. We did fix a but with sequence length in 10.1, so perhaps the issue was related to that.

**Comment by hanseoup** [comment]
Thank you for your reply! even after closing the issue :)
